src/Uno.UI/UI/Xaml/Media/Brush.wasm.cs

```diff
@@ -21,88 +23,110 @@ internal static IDisposable AssignAndObserveBrush(Brush b, ColorSetterHandler co
 					"AcrylicBrush is ** not ** supported by the AssignAndObserveBrush. "
 					+ "(Instead you have to use the AcrylicBrush.Subscribe().)"));
 				return Disposable.Empty;
 			}
-
-			if (b is ImageBrush)
+			else if (b is ImageBrush)
 			{
 				Application.Current.RaiseRecoverableUnhandledException(new InvalidOperationException(
 					"ImageBrush is ** not ** supported by the AssignAndObserveBrush. "
 					+ "(Instead you have to use the ImageBrush.Subscribe().)"));
 				return Disposable.Empty;
 			}
-
-			if (b is XamlCompositionBrushBase unsupportedCompositionBrush)
+			else if (b is XamlCompositionBrushBase unsupportedCompositionBrush)
 			{
-				// 
-				var disposables = new CompositeDisposable(2);
 				colorSetter(unsupportedCompositionBrush.FallbackColorWithOpacity);
 
-				unsupportedCompositionBrush.RegisterDisposablePropertyChangedCallback(
-						XamlCompositionBrushBase.FallbackColorProperty,
-						(s, colorArg) => colorSetter((s as XamlCompositionBrushBase).FallbackColorWithOpacity)
-					)
-					.DisposeWith(disposables);
-
-				unsupportedCompositionBrush.RegisterDisposablePropertyChangedCallback(
-						OpacityProperty,
-						(s, colorArg) => colorSetter((s as XamlCompositionBrushBase).FallbackColorWithOpacity)
-					)
-					.DisposeWith(disposables);
-
-				return disposables;
+				return WhenAnyChanged(new CompositeDisposable(2), unsupportedCompositionBrush, (s, e) => colorSetter((s as XamlCompositionBrushBase).FallbackColorWithOpacity), new[]
+				{
+					XamlCompositionBrushBase.FallbackColorProperty,
+					XamlCompositionBrushBase.OpacityProperty,
+				});
 			}
+			else
+			{
+				colorSetter(Colors.Transparent);
 
-			colorSetter(SolidColorBrushHelper.Transparent.Color);
-			return Disposable.Empty;
+				return Disposable.Empty;
+			}
 		}
 
 		// TODO: Refactor brush handling to a cleaner unified approach - https://github.com/unoplatform/uno/issues/5192
 		internal bool SupportsAssignAndObserveBrush => !(this is ImageBrush || this is AcrylicBrush);
-	}
 
+		private static IDisposable ObserveGradientBrushStops(GradientStopCollection stops, ColorSetterHandler colorSetter)
+		{
+			var disposables = new CompositeDisposable();
+			if (stops != null)
+			{
+				colorSetter(Colors.Transparent);
+
+				stops.VectorChanged += OnVectorChanged;
+				disposables.Add(() => stops.VectorChanged -= OnVectorChanged);
+			}
+
+			return disposables;
+
+			void OnVectorChanged(IObservableVector<GradientStop> sender, IVectorChangedEventArgs e)
+			{
+				colorSetter(Colors.Transparent);
+				foreach (var stop in stops.ToArray())
+				{
+					WhenAnyChanged(disposables, stop, (s, e) => colorSetter(Colors.Transparent), new[]
+					{
+						GradientStop.ColorProperty,
+						GradientStop.OffsetProperty,
+					});
+				}
+			}
+		}
+
+		private static CompositeDisposable WhenAnyChanged<T>(CompositeDisposable disposables, T source, PropertyChangedCallback callback, params DependencyProperty[] properties) where T : DependencyObject
```

Comment:
same comment here, can just be `Brush` instead of `T`

Reply:
Changed to `DependencyObject`, since one of them is a `GradientStop`.